LoRaEngine/modules/LoRaWanNetworkSrvModule/LoRaWan.NetworkServer/BasicsStation/ModuleConnection/ILoraModuleClient.cs

```diff
@@ -0,0 +1,26 @@
+// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+namespace LoRaWan.NetworkServer.BasicsStation.ModuleConnection
+{
+    using Microsoft.Azure.Devices.Client;
+    using Microsoft.Azure.Devices.Shared;
+    using System;
+    using System.Threading;
+    using System.Threading.Tasks;
+
+    public interface ILoraModuleClient : IAsyncDisposable
+    {
+        /// <summary>
+        /// Operation Timeout for the module connection in milliseconds.
+        /// </summary>
+        public TimeSpan OperationTimeout { get; set; }
+
+        public Task CreateFromEnvironmentAsync(ITransportSettings[] settings);
```

Comment:
Looking at the implementation and use, this seems to be a factory method that belongs on an interface like `ILoraModuleClientFactory`:

```c#
public interface ILoraModuleClientFactory
{
    Task<ILoraModuleClient> CreateFromEnvironmentAsync(ITransportSettings[] settings);
}
```

Reply:
I could move it to another class but then I would need to inject the Factory and the Class? I thought it was simple with a factory method like approach, but let's discuss tommorow maybe there is another approach